src/server/update_check.ts

```diff
@@ -0,0 +1,101 @@
+import { colors, join } from "./deps.ts";
+import { DEBUG } from "./constants.ts";
+
+interface CheckFile {
+  last_checked: string;
+  latest_version: string;
+  current_version: string;
+}
+
+async function getDenoDir(): Promise<string | undefined> {
+  const output = await new Deno.Command("deno", { args: ["info"] }).output();
+  const denoDir = colors.stripColor(new TextDecoder().decode(output.stdout))
+    .split(
+      /\n/g,
+    ).find(
+      (line) => line.startsWith("DENO_DIR"),
+    )?.replace("DENO_DIR location: ", "");
+
+  return denoDir;
+}
+
+async function fetchLatestVersion() {
+  const res = await fetch("https://dl.deno.land/fresh/release-latest.txt");
+  if (res.ok) {
+    return (await res.text()).trim().replace(/^v/, "");
+  }
+
+  throw new Error(`Could not fetch latest version.`);
+}
+
+export async function updateCheck(
+  interval: number,
+  getLatestVersion = fetchLatestVersion,
+) {
+  // Skip update checks on CI or Deno Deploy
+  if (
+    Deno.env.get("CI") === "true" || Deno.env.get("FRESH_NO_UPDATE_CHECK") ||
+    !DEBUG
+  ) {
+    return;
+  }
+
+  // Abort if we couldn't find a deno_dir
+  const denoDir = await getDenoDir();
+  if (!denoDir) {
+    return;
+  }
+
+  const versions = (await import("../../versions.json", {
```

Comment:
Is this portable?

Reply:
What do you mean with "portable"?

Reply:
Ah sorry, I misread this as a fetch. Closing this.